File: Problemset/running-sum-of-1d-array/running-sum-of-1d-array.java

// @Title: 一维数组的动态和 (Running Sum of 1d Array)
// @Author: 846188037@qq.com
// @Date: 2020-07-14 21:33:14
// @Runtime: 0 ms
// @Memory: 39.2 MB

class Solution {
    public int[] runningSum(int[] nums) {
     for(int i=1;i<nums.length;i++){
         nums[i]+=nums[i-1];
     }
     return nums;
    }
}

File: Problemset/running-sum-of-1d-array/running-sum-of-1d-array.py

# @Title: 一维数组的动态和 (Running Sum of 1d Array)
# @Author: 846188037@qq.com
# @Date: 2020-07-14 20:41:03
# @Runtime: 68 ms
# @Memory: 13.3 MB

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        runningSum=list()
        for i in range(len(nums)):
            runningSum.append(sum(nums[:i+1]))
        return runningSum
